parse.go

```diff
@@ -13,6 +13,9 @@ func rewriteRange(i string) string {
 	}
 	o := i
 	for _, v := range m {
+		if strings.HasPrefix(v[0], "v") && versionRegex.MatchString(v[0]) {
+			continue
```

Comment:
I'm not sure about the method of detecting a leading `v`. The leading `v` is not part of the [SemVer spec](http://semver.org). Applications other than dep import this package and don't require the `v` the way the go community wants to.

Any ideas on another method? The 1.x version looks for a space on both sides of the - for a range.

Reply:
Apologies for letting this drop off my radar entirely!

Would you be up for requiring that ranges must have a space around the dash? Right now the v2 branch has tests that support `2-3` which made it nearly impossible for me to do what I needed without the leading v trick. If we went back to requiring spaces for a range, then it wouldn't be a problem.

Reply:
@carolynvs sorry this dropped of my radar. If there was a space surrounding the dash it would work for me. The currently implementation on the v1 branch requires one or more space characters on each side of the dash.

This would be so much easier if there was a spec for ranges.